participant/models.py

```diff
@@ -85,19 +89,28 @@ def phishing_email_behaviour_and_actions(self, email_id):
         """
 
         def participant_behaviour(actions):
-            pb = ParticipantBehaviour.NEUTRAL
-            if [a for a in actions if a.get('action_type') in POSITIVE_ACTIONS]:
-                pb = ParticipantBehaviour.POSITIVE
-            if [a for a in actions if a.get('action_type') in NEGATIVE_ACTIONS]:
-                pb = ParticipantBehaviour.NEGATIVE
+            behaviour = ParticipantBehaviour.NEUTRAL
+            action_id = None
+
+            for a in actions:
+                if a[1].get("action_type") in POSITIVE_ACTIONS:
+                    behaviour = ParticipantBehaviour.POSITIVE
+                    action_id = a[0]
+
+                if a[1].get("action_type") in NEGATIVE_ACTIONS:
+                    behaviour = ParticipantBehaviour.NEGATIVE
+                    action_id = a[0]
+                    break
+
+            pb = {"behaviour": behaviour, "action_id": action_id}
             return pb
 
         actions = []
 
         if email_id in self.exercise.phishing_email_ids:
             for action_id, action_details in self.actions.items():
-                if email_id == action_details.get('email_id'):
-                    actions.append(action_details)
+                if email_id == action_details.get("email_id"):
+                    actions.append([action_id, action_details])
```

Comment:
Please include action_id in details instead as below:
```suggestion
                    action_details["action_id"] = action_id
                    actions.append(action_details)
```

Reply:
Good. Avoided it because it was adding extra "action_id" item in all `participantActions` objects (in endpoint response).